packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx

```diff
@@ -1,29 +1,71 @@
-import { forwardRef } from 'react';
+import { forwardRef, useCallback } from 'react';
 
-import { useCoreOrganization, useCoreUser, useOrganizationSwitcherContext } from '../../contexts';
-import { Button, descriptors, Icon, localizationKeys } from '../../customizables';
+import {
+  useCoreOrganization,
+  useCoreOrganizationList,
+  useCoreUser,
+  useOrganizationSwitcherContext,
+} from '../../contexts';
+import { Box, Button, descriptors, Icon, localizationKeys, NotificationBadge } from '../../customizables';
 import { OrganizationPreview, PersonalWorkspacePreview, withAvatarShimmer } from '../../elements';
+import { useDelayedVisibility, usePrefersReducedMotion } from '../../hooks';
 import { Selector } from '../../icons';
-import type { PropsOfComponent } from '../../styledSystem';
+import type { PropsOfComponent, ThemableCssProp } from '../../styledSystem';
+import { animations } from '../../styledSystem';
+import { organizationListParams } from './utils';
 
 type OrganizationSwitcherTriggerProps = PropsOfComponent<typeof Button> & {
   isOpen: boolean;
 };
 
+function useEnterAnimation() {
+  const prefersReducedMotion = usePrefersReducedMotion();
+
+  const getFormTextAnimation = useCallback(
+    (enterAnimation: boolean): ThemableCssProp => {
+      if (prefersReducedMotion) {
+        return {
+          animation: 'none',
+        };
+      }
+      return t => ({
+        animation: `${enterAnimation ? animations.notificationAnimation : animations.outAnimation} ${
+          t.transitionDuration.$textField
+        } ${t.transitionTiming.$slowBezier} 0s 1 normal forwards`,
+      });
+    },
+    [prefersReducedMotion],
+  );
+
+  return {
+    getFormTextAnimation,
+  };
+}
+
 export const OrganizationSwitcherTrigger = withAvatarShimmer(
   forwardRef<HTMLButtonElement, OrganizationSwitcherTriggerProps>((props, ref) => {
+    const { getFormTextAnimation } = useEnterAnimation();
     const { sx, ...rest } = props;
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const { username, primaryEmailAddress, primaryPhoneNumber, ...userWithoutIdentifiers } = useCoreUser();
     const { organization } = useCoreOrganization();
     const { hidePersonal } = useOrganizationSwitcherContext();
 
+    /**
+     * Prefetch user invitations and suggestions
+     */
+    const { userInvitations, userSuggestions } = useCoreOrganizationList(organizationListParams);
+
+    const notificationCount = (userInvitations.count ?? 0) + (userSuggestions.count ?? 0);
```

Comment:
```suggestion
    const notificationCount = (userInvitations.count || 0) + (userSuggestions.count || 0);
```

as we don't really care about `false`, `undefined` or `null` values here